src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = {};
+
+  Object.assign(copyState, state);
+
+  const arrayOfStates = [];
+
+  for (const action of actions) {
+    switch (action['type']) {
+      case 'addProperties':
+        Object.assign(copyState, action['extraData']);
+        break;
+
+      case 'removeProperties':
+        for (const property of action['keysToRemove']) {
+          delete copyState[property];
+        }
+        break;
+
+      case 'clear':
+        for (const key in copyState) {
+          delete copyState[key];
+        }
```

Comment:
```suggestion
       copyState = {}
```

Reply:
you don't need for there, you can just set copyState = {}